appinventor/components/src/com/google/appinventor/components/runtime/Spinner.java

```diff
@@ -55,16 +66,274 @@ public Spinner(ComponentContainer container) {
     adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
     view.setAdapter(adapter);
     view.setOnItemSelectedListener(this);
+    //Aayush
+    Enabled(true);
+      FontBold(bold);
+      FontTypeface(fontTypeface);
+      FontItalic(italic);
+      FontSize(fontSize);
+      BackgroundColor(backgroundColor);
+      TextColor(textColor);
+      TextAlignment(textAlignment);
 
     container.$add(this);
   }
 
+  /**
+   * Returns the Spinner's background color as an alpha-red-green-blue
+   * integer.
+   *
+   * @return  background RGB color with alpha
+   */
+  @SimpleProperty(
+          category = PropertyCategory.APPEARANCE,
+          description = "Returns the button's background color")
+  public int BackgroundColor() {
+    return backgroundColor;
+  }
+
+  /**
+   * Specifies the Spinner's background color as an alpha-red-green-blue
+   * integer.  If the parameter is {@link Component#COLOR_DEFAULT}, the
+   * original beveling is restored.  If an Image has been set, the color
+   * change will not be visible until the Image is removed.
+   *
+   * @param argb background RGB color with alpha
+   */
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_COLOR,
+          defaultValue = Component.DEFAULT_VALUE_COLOR_DEFAULT)
+  @SimpleProperty(description = "Specifies the button's background color. " +
+          "The background color will not be visible if an Image is being displayed.")
+  public void BackgroundColor(int argb) {
+    backgroundColor = argb;
+      //if you want to change only background of tesxtbox color
+    if(selectionIndex==0) return;
+    TextView selectedTextView = (TextView) view.getSelectedView();
+    TextViewUtil.setBackgroundColor(selectedTextView, argb);
+  }
+
   @Override
   public View getView(){
     return view;
   }
 
   /**
+   * Returns true if the Spinner is active and clickable.
```

Comment:
Are most of these functions very similar to functions in Button or ButtonBase? Any chance to extract them out to a collaboration class and use them in all classes needed?